### ISSUE TITLE
Continuous Functions Module

### ISSUE THREAD

**Issue by MarcelloPoletti** [highcharts/highcharts#21013]
Hi,
There are many cases where there is a need to represent continuous trends rather than discrete ones. Some examples include:

- Trend lines, which are not necessarily linear.
- Line annotations, which are not necessarily straight lines.

These types of series or annotations are essentially continuous functions rather than a set of points. The topic is evidently complex and broad. There are challenges in defining the domain and codomain of functions, and functions can be expressed in various ways - explicitly, parametrically, and so on.

However, by imposing certain restrictions (at least initially), we could create a powerful tool. For instance, we could consider only:

- Explicit functions, where y = f(x).
- A domain interval, possibly open (e.g., [-∞, +∞]).
- An optional density parameter that specifies the drawing precision (perhaps unnecessary?).
- 
A series or annotation associated with a function could then be drawn by the engine. Naturally, this leads to some special behaviors:

A function-based series would not have markers.
A function-based series would not automatically define any axis visibility range.
Crosshairs, tooltips, etc., would require careful consideration.
In summary, this is indeed a non-trivial task, but if someone willing takes it up, it would benefit the entire community! 😊

Marc.

**Comment by karolkolodziej** [comment]
Hi @MarcelloPoletti! 

Thank you for sharing this idea! 

Right now with a bit of code, you can create such a thing- [see the y = f(x) demo.](https://jsfiddle.net/BlackLabel/b0Lqmnv6/)

But if I understand you correctly your idea is to have instead of the `data` in the series something like this:
```js
  series: [{
    type: 'function',
    equation: 'x^2 + 5'
    range: [-10, 10]
  }]
```

If you have any other idea how it might look or work please feel free to share that!

**Comment by MarcelloPoletti** [comment]
Hi [karolkolodziej](https://github.com/karolkolodziej),

Yes, a function type series it's probably the best way. And it can have some params, in example:

`series: [{
    type: 'function',
    equation: function(x){ return x^2 + 5},
    range /*optional*/: [-10, 10], //or [-10, null] or [null, 10]...
    density /*optional*/: 10, //or nothing, draw a point every 10 pixel 
  }]`

marc.

**Comment by karolkolodziej** [comment]
Thank you for clarifying! We are going to discuss it internally and we will keep you posted!
